dm/projects.go

```diff
@@ -0,0 +1,117 @@
+package dm
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type ProjectDetails struct {
+	Details DataDetails `json:"details, omitempty"`
+}
+
+// ListBuckets returns a list of all buckets created or associated with Forge secrets used for token creation
+func (api HubAPI) ListProjects(hubKey string) (result HubDetails, err error) {
+	
+	// TO DO: take in optional arguments for query params: id, ext, page, limit
+	// https://forge.autodesk.com/en/docs/data/v2/reference/http/hubs-hub_id-projects-GET/
+	bearer, err := api.Authenticate("data:read")
+	if err != nil {
+		return
+	}
+
+	path := api.Host + api.HubAPIPath
+
+	return listProjects(path, hubKey, "", "", "", "", bearer.AccessToken)
+}
+
+func (api HubAPI) GetProjectDetails(hubKey, projectKey string) (result ProjectDetails, err error) {
+	bearer, err := api.Authenticate("data:read")
+	if err != nil {
+		return
+	}
+	path := api.Host + api.HubAPIPath
+
+	return getProjectDetails(path, hubKey, projectKey, bearer.AccessToken)
+}
+
+
+/*
+ *	SUPPORT FUNCTIONS
+ */
+func listProjects(path, hubKey, id, extension, page, limit string, token string) (result HubDetails, err error) {
```

Comment:
Might be great to break out the query params. I liked how Contentul's go sdk did it (as an example)

https://github.com/contentful-labs/contentful-go/blob/master/query.go

Reply:
per chat, punting! captured in trello ticket